backends/platform/dc/dclauncher.cpp

```diff
@@ -0,0 +1,127 @@
+#define FORBIDDEN_SYMBOL_EXCEPTION_strcasecmp
+
+#include "DCLauncherDialog.h"
+
+#include "dc.h"
+
+#include "common/translation.h"
+#include "common/system.h"
+#include "common/config-manager.h"
+#include "gui/gui-manager.h"
+#include "gui/massadd.h"
+#include "gui/browser.h"
+#include "gui/launcher.h"
+
+#include "base/plugins.h"
+
+#include "icon.h"
+
+#include <ronin/gddrive.h>
+
+extern Icon icon;
+
+DCLauncherDialog::DCLauncherDialog()
+	: LauncherDialog(), cdWasOpen(false)
+{
+	Common::String lastpath;
+
+	PluginManager::instance().loadAllPlugins();
+#if defined(DYNAMIC_MODULES)
+	if (PluginManager::instance().getPlugins(PLUGIN_TYPE_ENGINE).empty()) {
+		GUI::BrowserDialog browser(_("Select plugin directory"), true);
+		lastpath = ConfMan.get("browser_lastpath");
+		ConfMan.set("browser_lastpath", "/plugins");
+		if (browser.runModal() > 0) {
+			Common::FSNode result = browser.getResult();
+			((OSystem_Dreamcast *)g_system)->setPluginCustomDirectory(result.getPath());
+			PluginManager::instance().loadAllPlugins();
+		}
+		ConfMan.set("browser_lastpath", lastpath);
+	}
+#endif
+	massAdd();
+}
+
+void DCLauncherDialog::massAdd()
+{
+	GUI::MassAddDialog massAddDlg(Common::FSNode(""), true);
+	massAddDlg.runModal();
+	// Update the ListWidget and force a redraw
+	updateListing();
+	g_gui.scheduleTopDialogRedraw();
+}
+
+void DCLauncherDialog::handleTickle()
+{
+	unsigned int param[4];
+	gdGdcGetDrvStat(param);
+
+	int cdState = param[0];
+	if (cdState >= 6 && !cdWasOpen) {
+		cdfs_reinit();
+		updateListing();
+		g_gui.scheduleTopDialogRedraw();
+		cdWasOpen = true;
+	}
+	else if (cdState > 0 && cdState < 6 && cdWasOpen) {
+		cdWasOpen = false;
+		cdfs_reinit();
+		massAdd();
+	}
+}
+
+static bool isIcon(const Common::FSNode &entry)
+{
+	int l = entry.getDisplayName().size();
```

Comment:
Previously, commit b224b59 changed this to use hasSuffixIgnoreCase(). Is there any reason why this was changed back?

Reply:
I don't know if there was a particular reason, but it doesn't really matter if the comparison is done ignoring case or not; filenames on ISO-9660 are always uppercase...